vpc/tool/container2/setup_container_linux.go

```diff
@@ -259,144 +258,137 @@ func addIPv6AddressAndRoutes(ctx context.Context, allocation types.Allocation, n
 	runtime.LockOSThread()
 	defer runtime.UnlockOSThread()
 
-	// Amazon only gives out /128s
-	new6IP := net.IPNet{IP: net.ParseIP(allocation.IPV6Address.Address.Address), Mask: net.CIDRMask(128, 128)}
-	new6Addr := netlink.Addr{
-		// TODO (Sargun): Check IP Mask setting.
-		IPNet: &new6IP,
-		Flags: unix.IFA_F_PERMANENT | unix.IFA_F_NODAD,
-		Scope: unix.RT_SCOPE_UNIVERSE,
-	}
-	err := nsHandle.AddrAdd(link, &new6Addr)
-	if err != nil {
-		logger.G(ctx).WithError(err).Error("Unable to add IPv6 addr to link")
-		return errors.Wrap(err, "Unable to add IPv6 addr to link")
-	}
-
-	var linkLocalAddr netlink.Addr
-	for {
-		// We need *a* permanent address
-		addrs, err := nsHandle.AddrList(link, netlink.FAMILY_V6)
-		if err != nil {
-			return fmt.Errorf("Cannot get addrs: %w", err)
-		}
-
-		for _, addr := range addrs {
-			if addr.Scope == unix.RT_SCOPE_LINK {
-				linkLocalAddr = addr
-				goto out
-			}
-		}
-		select {
-		case <-ctx.Done():
-			return fmt.Errorf("Did not find permanent link-local address in set: %v: %w", addrs, ctx.Err())
-		case <-time.After(100 * time.Millisecond):
-		}
-	}
-
-out:
 	origns, err := netns.Get()
-	defer origns.Close()
 	if err != nil {
 		return fmt.Errorf("Could not get current net ns: %w", err)
 	}
+	defer origns.Close()
 
 	err = netns.Set(netns.NsHandle(netnsfd))
 	if err != nil {
 		return fmt.Errorf("Could not switch net ns: %w", err)
 	}
 
-	networkInterface, err := net.InterfaceByName("eth0")
-	if err != nil {
-		return fmt.Errorf("Could not get network interface eth0: %w", err)
+	var mtu *uint32
+	var routes []struct {
+		mtu   uint32
+		route net.IPNet
 	}
+	for _, route := range allocation.Routes {
+		if route.Family != vpcapi.AssignIPResponseV3_Route_IPv6 {
+			continue
+		}
 
-	conn, _, err := ndp.Dial(networkInterface, ndp.Addr(linkLocalAddr.IP.String()))
-	if err != nil {
-		return fmt.Errorf("Unable to dial NDP conn: %w", err)
+		_, routeNet, err := net.ParseCIDR(route.Destination)
+		if err != nil {
+			logger.G(ctx).WithError(err).Error("Could not parse route")
+			return fmt.Errorf("Could not parse route CIDR (%s): %w", route.Destination, err)
+		}
+
+		ones, zeros := routeNet.Mask.Size()
+		if routeNet.IP.Equal(net.IPv6zero) && ones == 0 && zeros == 128 {
+			mtu = &route.Mtu
```

Comment:
what is `mtu` in this context here, the default route mtu? if so can you rename the variable?